src/Index.php

```diff
@@ -381,26 +381,31 @@ public function refresh(): Response
      */
     public function create(array $args = [], $options = null): Response
```

Comment:
General thought: Should we become more strict on what is allowed to be passed to options? The code below is way too complex as we piled onto it over to time not have a breaking change I guess. Good news is, your PR seems to simplify it at least.

Reply:
Update: Just read your PR message again. ++ on deprecation additions.

Reply:
Yes exactly, I will deprecate all types except array. So this will be backward compatible and will be easy to identify which part to drop when bumping a new major version.

Reply:
By the way there's this PR #1823 that is tied to deprecations, would be nice if you could have a look so that we can go forward and use this mechanism for all futures deprecations. I have several ones in mind ;)